src/Components/AvilableRooms/AvilableRooms.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useState, useEffect, useContext } from "react";
+import axios from "axios";
+import {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  Paper,
+  TablePagination,
+  IconButton,
+  Grid,
+  Card,
+  CardMedia,
+  Typography,
+  CardActionArea,
+} from "@mui/material";
+import { Navigate, useLocation, useNavigate, useParams } from "react-router-dom";
+import EventAvailableIcon from '@mui/icons-material/EventAvailable';
+import { Container } from "@mui/system";
+import FavoriteIcon from '@mui/icons-material/Favorite';
+import InfoIcon from '@mui/icons-material/Info';
+import { AuthContext } from "../../Context/AuthContext";
+import DetailsRoom from "../RoomDetails/RoomDetails";
+import { format } from "date-fns";
+
+function AvailableRooms() {
+//   const { startDate, endDate } = useParams();
+  const [roomsList, setRoomsList] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pagesArray, setPagesArray] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const navigate =useNavigate()
+  const [hoveredIndex, setHoveredIndex] = useState(null);
+  const [favoriteRooms, setFavoriteRooms] = useState([]);
+  const{baseUrl,reqHeaders}=useContext(AuthContext)
+  const location = useLocation();
+const count=location?.state?.count
+const dateRange=location?.state?.ranges
+const startDate=`${format(dateRange[0].startDate,'ddMMM')}`
+const endDate=`${format(dateRange[0].endDate,'ddMMM')}`
+console.log(startDate,endDate);
+
+  const fetchAvailableRooms = (page) => {
+    axios
+      .get(`http://154.41.228.234:3000/api/v0/portal/rooms/available`, {
+        params: {
+          page,
+          size: rowsPerPage,
+          startDate,
+          endDate,
+        },
+      })
+      .then((response) => {
+        console.log(response);
+        
+        setPagesArray(response?.data?.data?.totalCount || 0);
+        setRoomsList(response.data.data.rooms);
+      })
+      .catch((error) => {
+        console.error("Error fetching available rooms:", error);
+      });
+  };
+
+  const addToFavorites = (roomId) => {
```

Comment:
Consider adding toast messages to indicate whether a room is added or removed from favorite.